posydon/binary_evol/DT/step_detached.py

```diff
@@ -696,6 +696,49 @@ def get_mist0(self, star, htrack):
 
                     x0, method="TNC", bounds=bnds,
                     )
+                # match a pro-MT postMS star to a He star
+                if (np.abs(sol.fun) > tolerance_mist_integration
+                        or not sol.success):
+                    star.htrack = False
+                    MESA_label = ["he_core_mass", "center_he4", "log_R"]
+                    posydon_attribute = [star.he_core_mass,
+                                         star.center_he4, star.log_R]
+                    rs = [10.0, 1.0, 2.0]
+                    if self.verbose:
+                        print("Matching attributes and their normalizations : ",
+                              MESA_label, rs)
+                    for i in MESA_label:
+                        if i not in self.root_keys:
+                            raise Exception("Expected matching parameter not added"
+                                            " in the MIST model options.")
+                    x0 = get_root0(
+                        MESA_label, posydon_attribute, star.htrack, rs=rs)
+                    bnds = ([m_min_He, m_max_He], [0, None])
+                    sol = minimize(
+                        lambda x: (
+                            (sc_he_core_mass_He.transform(
+                                get_track_val(MESA_label[0], htrack, *x))
```

Comment:
Should it be "star.htrack" instead of "htrack" here?

Reply:
There must be a better way to organize it. Right now the htrack here doesn't affect the matching because the function is called twice latter. The first time star.track becomes False and the returned solution is not used. Then the second  time htrack becomes the same with star.htrack and the solution is used.

Reply:
Because we have two independent variables `htrack` and `star.htrack`, there should be a reason for them to not be the same. Hence, please check when they are not to same and which one is the one which should be used here in the future.
I get to the question, because in the rest of your changes you use `star.htrack`, thus in terms of consistency I was wondering, whether there is a reason to not use it here but use `htrack` instead.